### ISSUE TITLE
New groups are not applied on the Remote-SSH session without rebooting the remote machine

### ISSUE THREAD

**Issue by ElleshaHackett** [microsoft/vscode-remote-release#5813]
As far as I can tell these two closed issues (https://github.com/microsoft/vscode-remote-release/issues/1997, https://github.com/microsoft/vscode-remote-release/issues/4058) still apply and this is annoying as hell as it took me a lot of time to figure out that the issue was VScode and not my remote setup. Below steps are 100% reproducable.

- VSCode Version: 1.61.2
- Local OS Version: Windows 10 21H1
- Remote OS Version: Ubuntu 20.04.3 LTS
- Remote Extension/Connection Type: Visual Studio Code Remote - SSH extension (https://code.visualstudio.com/docs/remote/ssh)
- Logs: n/a

**Description**:
I am scoping user folders for different Docker containers. **Note that Docker has nothing to do with this issue.** For this reason I am creating groups and users on my remote Ubuntu machine and applying them to specific folders. I also add the user that I used for Remote SSH in VScode to the newly created groups. And this change is not applied, also not after adding it manually to the current session, after reconnecting and after reopening VScode altogether. The only thing that seems to fix it is to reboot my remote machine. I did not find a solution for this other then rebooting the whole server, which seems a bit dramatic.

**Steps to Reproduce (on the remote machine):**
For this example let's assume the user I use to connect to the server over Remote-SSH in VScode is `remoteUser` and the remote folder I am trying to access as that user is `/tmp/test`.

For the sake of completeness, I use the following permissions on that (parent) folder:

```
sudo chown root:root /tmp
sudo find /tmp -type d -exec chmod 0775 {} +
sudo find /tmp -type f -exec chmod 0660 {} +
sudo chown -R 5000:5000 /tmp/test
sudo find /tmp/test -type d -exec chmod 0770 {} + 
sudo find /tmp/test -type f -exec chmod 0660 {} +
```

1. `sudo groupadd --gid 5000 newDockerGroup`
2. `sudo usermod -a -G newDockerGroup remoteUser` (adding `remoteUser` to the newly created group)
3. `groups remoteUser` viewing the current groups `remoteUser` is member of. The result does contain `newDockerGroup` as expected.
4. In the vscode terminal on my current session as remoteUser I run: `id -Gn` and it **does not** contain `newDockerGroup`. This is unexpected behaviour.
5. I run `newgrp newDockerGroup`. Now after checking `id -Gn` again as user remoteUser, it does contain `newDockerGroup`. This is expected. Now the user is applied on the current Terminal session and I can browse, open and write in that folder as user `remoteUser` using Terminal. But the Remote-SSH session isn't changed and the file explorer in vscode does not allow me to navigate that folder. Reconnecting the session, reopening vscode does not fix this issue and will return to a state where `id -Gn` results in an answer **without the newly generated group**. The only way so far to fix this is rebooting the entire remote machine. After which everything works find in vscode.

_I tried restarting the remote SSH service without success. If there is another workaround to restart specific processes (remotely) to get this fixed I would love to know because restarting the whole remote host is not an option._

_Moreover I open VSCode and from the Remote Explorer tab in vscode I open a new window which allows me to browse the remote files in the Explorer tab._

**Comment by roblourens** [comment]
Does it work if you run the "Kill VS Code Server on Host" command to restart the VS Code server?

**Comment by ElleshaHackett** [comment]
Running this option **does not** solve it:
![image](https://user-images.githubusercontent.com/5116641/140033505-0b9c295e-f271-4662-84a8-e127779d8bf9.png)

Then running this option "close remote connection" and reconnecting **does not** solve it either:
![image](https://user-images.githubusercontent.com/5116641/140033671-8d2aa558-2fc9-437f-a903-f23b6151cc82.png)

Then closing all VScode locally and reopening the entire application, reconnecting **does not** solve it either.

_Moreover I wonder if any "VS Code Server" is even installed on the host. Would that be done automatically? I just connected to a remote SSH with my local vscode and as far as I know I didn't specifically install anything on the server._

---

**Edit:** I did additional research and found this: https://stackoverflow.com/questions/56892931/how-to-kill-vscode-remote-services-on-ubuntu-host. After running `ps uxa | grep .vscode-server` on the remote host, I could see multiple (6) processes running. So apparently its installed and running remotely. However after killing those processes manually using `ps uxa | grep .vscode-server | awk '{print $2}' | xargs kill -9` _(thanks user "inselberg" as StackOverflow)_, I did get prompted in vscode to reconnect (which did not happen after using the "Kill VS Code Server on Host" command). After manually killing the Remote VS Code Server and reconnecting locally, the new permissions where finally applied and Explorer starts to function properly again.

However, this issue still remains, it's unexpected behavior even after a reconnect. Moreover there should be a new issue created here because apparently the "Kill VS Code Server on Host" command doesn't work either.

**Comment by roblourens** [comment]
I replied in the other thread, but that command should have the same effect as what you run. But yes, you have to restart the server to pick up changes to the environment like this.

Any suggestion for making this more obvious?

**Comment by Tracnac** [comment]
Hi,

>> Any suggestion for making this more obvious?

Yes why not closing vscode-server on the remote host when quitting vscode/your session !?
Is there a reason to let it run at the end of your session ?

**Comment by roblourens** [comment]
It makes startup faster for subsequent connections. For example, if you open a new folder and the window reloads, that is like a "new connection" and should be fast.

**Comment by ElleshaHackett** [comment]
Please reopen. All the information is supplied.

**Comment by tanhakabir** [comment]
@ElleshaHackett just to verify after using Remote-SSH Nightly and using the Kill command to kill the remote didn't solve your issue?

**Comment by ElleshaHackett** [comment]
Please reopen. @tanhakabir I did not have time to verify this yet. I will try to do this as soon as I have time.

**Comment by acalvino4** [comment]
Why is the auto-close configured to 8 days?  If the issue is waiting on a relevant person, they could be on vacation for twice that length of time an never even see a comment till the thread is already closed. Or, given that much of this is open-source effort, he relevant person could just be busy on their main gig for a month or more and just not have a chance to address what's needed. I get not wanting issues to be sitting open for months on end, but 8 days just seems ridiculous.
